### ISSUE TITLE
Decryption validation error

### ISSUE THREAD

**Issue by earthlingworks** [gettalong/hexapdf#138]
We're sometimes seeing this error come up: 

HexaPDF::EncryptionError: Invalid data for decryption, need 32 + 16*n bytes

We monkey patched the class and commented these lines (and everything works great): 

```
# lib/ hexapdf/encryption/aes.rb 143:13
if data.length < BLOCK_SIZE || data.length % BLOCK_SIZE != 0
  raise HexaPDF::EncryptionError, "Invalid data for decryption, need 32 + 16*n bytes"
end
```

We tried `validate: false` but that didn't bypass this validation.

**Comment by earthlingworks** [comment]
Almost forgot the PDF: you can download one of the PDFs that gets this error from here: https://share.getcloudapp.com/4gu2rqkG

Thanks!

**Comment by gettalong** [comment]
Thanks for reporting!

I tried decrypting the provided file using `hexapdf mod --strict --decrypt /tmp/e2249a99-929a-4118-ac1f-038ed1d836cd.pdf /tmp/out.pdf` but no error is thrown.

Could you describe what you are doing when this error is thrown? Or provide a small sample script that reproduces the problem?

**Comment by earthlingworks** [comment]
Sorry about the delay (was in the middle of a few things). Looks like part of the problem was that we had the wrong PDF, so here's the one that's giving problems (same just a bit different): https://share.getcloudapp.com/o0u6Ydqv

Had some trouble duplicating with a simple script, but we have it now:

      pdf = HexaPDF::Document.new 
      pdf2 = HexaPDF::Document.new(io: open('/path-to-file/decryption-error2.pdf'))

      importer = HexaPDF::Importer.for source: pdf2, destination: pdf
      pdf2.pages.each do |page|
        pdf.pages << importer.import(page)
      end

      pdf.write('/path-to-file/decrypt-error-out.pdf')

**Comment by gettalong** [comment]
@earthlingworks Thanks for the file and the code! I have found the problematic part and will now look for a good solution.

**Comment by gettalong** [comment]
@earthlingworks I have just released version 0.15.3 with a fix for the issue encountered with the file in https://share.getcloudapp.com/o0u6Ydqv

Since you mentioned *"...one that's giving problems (same just a bit different)..."*, please let me know if there are still issues.

One funny thing with this is that the file was generated by Acrobat Distiller and that it probably **doesn't create a valid PDF file**. The spec says with regards to encrypting metadata streams and the `/EncryptMetadata` key: *"Indicates whether the document-level metadata stream shall be encrypted."*. But in the case of this file a component level metadata stream also wasn't encrypted.
